lib/screens/home/home_repository.dart

```diff
@@ -0,0 +1,75 @@
+import 'dart:convert';
+import 'dart:math';
+
+import 'package:ecellapp/core/res/errors.dart';
+import 'package:ecellapp/models/user.dart';
+import 'package:flutter/material.dart';
+import 'package:ecellapp/core/res/strings.dart';
+import 'package:ecellapp/core/utils/injection.dart';
+import 'package:ecellapp/core/utils/logger.dart';
+import 'package:http/http.dart' as http;
+import 'package:shared_preferences/shared_preferences.dart';
+
+@immutable
+abstract class HomeRepository {
+  /// Takes in `token` , gives the user details and throws a suitable exception if something goes wrong.
+  Future<User> profile(String token);
+}
+
+class FakeHomeRepository implements HomeRepository {
+  @override
+  Future<User> profile(String token) async {
+    // Simulate network delay
+    await Future.delayed(Duration(seconds: 1));
+
+    if (Random().nextBool()) {
+      // random network error
+      throw NetworkException();
+    } else {
+      var json = {
+        S.firstnameKey: "Sahil",
+        S.lastnameKey: "Silare",
+        S.emailKey: "sahil@gmail.com",
+        S.phoneKey: "",
+      };
+      User user = User.fromJson(json);
+      // fake successful response (the data entered here same as in the API Doc example)
+      return user;
+    }
+  }
+}
+
+class APIHomeRepository implements HomeRepository {
+  final String classTag = "APIHomeRepository";
+  @override
+  Future<User> profile(String token) async {
+    String token = sl.get<SharedPreferences>().getString(S.tokenKey);
```

Comment:
`S.tokenKey` is supposed to be an api key, and not Shared Preferences key. Add a new key `strings.dart` for this usecase.

Reply:
done